If an invalid validation rule is added to validations, throw an error

luislobo opened the pull request:
For example, while testing, I changed from ip, to isIp (lowercase p) and it throwned an error, because isIp was not a ke of the VALIDATIONS array. This way, a proper error is thrown.

sgress454 commented:
@luislobo Seems valid to me, but I'm curious how you were getting this error.  How were you defining your model?  Were you declaring a `validations` dictionary directly?

luislobo commented:
I first had this:

```javascript
module.exports = {
  attributes: {
    name: {
      type: 'string',
      required: true
    },
    ip: {
      type: 'string',
      required: true,
      ip: true
    }
};

```
Sails said:
```
2017-03-23T17:45:32.457Z - debug: In attribute `ip` of  model `xxx`:
2017-03-23T17:45:32.459Z - debug:   The `ip` validation rule is now `isIP` (changing it for you this time).
```
So I changed it to `isIp` (note lowercase p), but got this:

```
error: Error: The attribute `ip` on the `xxx` model contains invalid properties. The property `isIp` isn't a recognized property.
```

Then I checked the new format, and added:
```javascript
ip: {
      type: 'string',
      required: true,
      validations: {
        isIp: true // <<<< notice still isIp is lowercase `p`
      }
    },
```
So I got this error
```
2017-03-23T17:50:02.352Z - error: A hook (`orm`) failed to load!
error: TypeError: Cannot read property 'expectedTypes' of undefined
  ```
Changing it to `isIP` made it work, but the error message was very misleading, so, I created this PR.
Now it says:
```
In the `ip` attribute of model `rcu`:
The `isIp` validation rule is not a supported validation rule.
Supported validation rules are: 
 - isBoolean
 - isNotEmptyString
 - isInteger
 - isNumber
 - isString
 - max
 - min
 - isAfter
 - isBefore
 - isCreditCard
 - isEmail
 - isHexColor
 - isIn
 - isIP
 - isNotIn
 - isURL
 - isUUID
 - minLength
 - maxLength
 - regex
 - custom
-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-
```

sgress454 commented:
> Then I checked the new format, and added:

That's what I mean -- is there somewhere in the docs where it says to put validations as a dictionary under a `validations` key?  Or did you just see that in the code?  We decided to just set up that `validations` dictionary under the hood rather than requiring the end-user to declare it, just to avoid another migration step. 

This is kind of a special case because it's possible you could find out about `validations` if you were looking at the Waterline standalone example (or tests), so I'm okay leaving your error as-is.  Otherwise I'd say we change it to a "consistency violation" error, since by the time this code runs, there should never be an invalid item in that `validations` dictionary.  Annnnnyway thank you and good work.
